src/interface/cmdTlmServer.c

```diff
@@ -0,0 +1,487 @@
+        isCmdValid = false;
+    }
+
+    if (!isCmdValid) {
+        pCmdStatus->header.frameId = FrameId_CmdStatus;
+        pCmdStatus->header.counter = pCmdMsg->counter;
+        pCmdStatus->cmdStatus = CmdStatus_NotOK;
+        pCmdStatus->duration = 0;
+        strncpy(&pCmdStatus->reason[0], reasonCmdFail,
+                LENGTH_CMD_STATUS_REASON);
+        pCmdStatus->reason[LENGTH_CMD_STATUS_REASON - 1] = '\0';
+    }
+
+    return isCmdValid;
+}
+
+// Run the server.
+// Note: The data types of input and output are required for pthread_create().
+// The input needs to cast to the correct data type.
+static void *cmdTlmServer_run(void *pData) {
+
+    // Get the server information
+    serverInfo_t *pServerInfo = (serverInfo_t *)pData;
+
+    // Wait until begins to run the server's job
+    while (!pServerInfo->isReady) {
+        sleep(1);
+    }
+
+    // Loop delay time
+    // 50 milliseconds (= 20 Hz)
+    struct timespec ts50;
+    ts50.tv_nsec = 50000000;
+    ts50.tv_sec = 0;
+
+    // Use poll() for the connected socket
+    struct pollfd fds;
+    fds.events = POLLIN;
+
+    // Wait for the connection to the server
+    syslog(LOG_NOTICE, "Waiting for the connection request in %s server.",
+           pServerInfo->pName);
+
+    // Run the server
+    int error;
+    while (pServerInfo->isReady) {
+
+        // New command message
+        commandStreamStructure_t cmdMsg;
+        commandStatusStructure_t cmdStatus;
+
+        // State machine to deal with the connection/disconnection with
+        // TCP/IP client
+        switch (pServerInfo->serverStatus) {
+
+        // Look for the connection with TCP/IP client
+        case ServerStatus_Disconnected:
+            // Wait for the connection request
+            pServerInfo->socketConnect = tcpServer_accept(
+                pServerInfo->socketListen, pServerInfo->timeout);
+            if (pServerInfo->socketConnect == -1) {
+                break;
+            } else {
+                pServerInfo->serverStatus = ServerStatus_Connected;
+
+                syslog(LOG_NOTICE,
+                       "The state of %s server is connected. socket = %d.",
+                       pServerInfo->pName, pServerInfo->socketConnect);
+
+                // Make use of poll() to receive the message
+                fds.fd = pServerInfo->socketConnect;
+            }
+            break;
+
+        // Connected with the TCP/IP client, look for commands
+        case ServerStatus_Connected:
+
+            // Reply the last command status from commanding.c
+            error = cmdTlmServer_sendLastCmdStateInMsgQueue(pServerInfo);
+
+            // We were connected but the connection has been broken
+            if (error < 0) {
+                cmdTlmServer_closeConn(pServerInfo, &fds);
+                break;
+            }
+
+            // Check the new command
+            error = cmdTlmServer_recv(&cmdMsg, &fds, pServerInfo->timeout);
```

Comment:
`error` is actually the number of bytes received. I strongly suggest using a clearer name, such as nbytes, because otherwise it is not clear to the reader how the code can ever get to line 359. I believe a conventional C error code is 0 for success (not the case here) and negative for error. Positive values aren't common.

Reply:
Use the variable `nbytes` in the commit: ccc826f. Thanks!